### ISSUE TITLE
Timeline error in epic "setIsIntervalData"

### ISSUE THREAD

**Issue by allyoucanmap** [geosolutions-it/MapStore2#8346]
## How to reproduce

-  login as admin
- open this map https://dev-mapstore.geosolutionsgroup.com/mapstore/#/viewer/openlayers/40116

*Expected Result*

We should fix the error and verify why is it not able to compute the result for the setIsIntervalData flow

*Current Result*

- error in console

```
Error in epic "setIsIntervalData". Original error: TypeError: Cannot read properties of null (reading 'indexOf')
```

- [x] Not browser related

<details><summary> <b>Browser info</b> </summary>

(use this site: <a href="https://www.whatsmybrowser.org/">https://www.whatsmybrowser.org/</a> for non expert users)

| Browser Affected | Version |
|---|---|
|Internet Explorer| |
|Edge| |
|Chrome| |
|Firefox| |
|Safari| |
</details>

**Comment by tdipisa** [comment]
@dsuren1 I need first of all an investigation to understand if this issue is a regression or not included after recent updates we provided for the timeline (eg. #7958).

**Comment by dsuren1** [comment]
@tdipisa 
This issue is a regression caused by this #8229 part of #7958 which introduced the new epics `setIsIntervalData`. 

Problem is caused because the `setIsIntervalData` epics is executed even when no layer is selected which is required to generate arguments for `getDomainValues` service.
Hence an `undefined` url is used to fire domain service call resulting in null values of next and previous time value causing the exception.

**Occurrences**
- On updating layer dimension data onMapConfigLoad
- On traversing time when `Snap to guide layer` is disabled

**Solution**
Allow `setIsIntervalData` epics execution only layer is selected  (timeline -> selectedLayer)

I will provide fix as part this issue itself.

**Comment by ale-cristofori** [comment]
@tdipisa @dsuren1  no, but I would  refactor the epic with two epics to keep concerns separated.

the one that listens for ` SELECT_LAYER` will get the `layerId` from the action and filter out those layers that do not have an ID, like in this case the layer has not been loaded correctly.

the one that listens for  `SET_CURRENT_TIME` will get the `layerId` from the state with `selectedLayerSelector`


https://github.com/geosolutions-it/MapStore2/blob/13d6dea3a769ba631c3c917668b166a5d172b268/web/client/epics/playback.js#L369

**Comment by dsuren1** [comment]
@ale-cristofori 
From what I can infer, the problem is caused only by `SET_CURRENT_TIME` when no layer is selected and the value of `selectedLayerSelector` is undefined. 
So in [domainArgs](https://github.com/geosolutions-it/MapStore2/blob/13d6dea3a769ba631c3c917668b166a5d172b268/web/client/epics/playback.js#L87) the value of `selectedLayerUrl` becomes undefined causing the error. Which can still happen with a separate epics.

How should we pick the layer to be used in this scenario, should we go with the first layer instead?

![image](https://user-images.githubusercontent.com/26929983/176114327-82721d1c-245d-48cb-81b7-242820e10c62.png)

**Comment by ale-cristofori** [comment]
@dsuren1 if the purpose is to avoid the request being fired with `undefined` layerId, shouldn't this cover both cases?

```javascript
export const setIsIntervalData = (action$, { getState = () => { } } = {}) =>
    action$.ofType(SET_CURRENT_TIME, SELECT_LAYER)
        .filter(({layerId}) => layerId || selectedLayerSelector(getState()))
        .switchMap(({time: actionTime}) => getTimeIsInterval(actionTime, getState));
```
But maybe I am missing something. Let me know

**Comment by dsuren1** [comment]
> @dsuren1 if the purpose is to avoid the request being fired with `undefined` layerId, shouldn't this cover both cases?
> 
> ```js
> export const setIsIntervalData = (action$, { getState = () => { } } = {}) =>
>     action$.ofType(SET_CURRENT_TIME, SELECT_LAYER)
>         .filter(({layerId}) => layerId || selectedLayerSelector(getState()))
>         .switchMap(({time: actionTime}) => getTimeIsInterval(actionTime, getState));
> ```
> 
> But maybe I am missing something. Let me know

@ale-cristofori 
That's exactly the solution I proposed in my comment [above](https://github.com/geosolutions-it/MapStore2/issues/8346#issuecomment-1167172777)  :)

Kindly let me know if I can proceed with it.
